apps/rpc/src/modules/core.ts

```diff
@@ -132,6 +134,20 @@ export async function createServiceLayer(
   clients: ReturnType<typeof createThirdPartyClients>,
   configuration: Configuration
 ) {
+  async function executeTransactionWithAudit<T>(
+    fn: (tx: typeof clients.prisma) => Promise<T>,
+    userId: string | null
+  ): Promise<T> {
+    return clients.prisma.$transaction(async (tx) => {
+      const transactionId = crypto.randomUUID()
+
+      await tx.$executeRaw`SELECT set_config('app.current_user_id', ${userId || null}, TRUE)`
+      await tx.$executeRaw`SELECT set_config('app.current_transaction_id', ${transactionId}, TRUE)`
+
+      return fn(tx as typeof clients.prisma)
+    })
```

Comment:
PostgreSQL has a transaction ID for every transaction. We should just use that instead.

Also, this cast is a little suspicious. Perhaps you should see if these types are sound (aka no assertions)


```suggestion
    fn: (tx: DBHandle) => Promise<T>,
    userId: UserId| null
  ): Promise<T> {
    return clients.prisma.$transaction(async (tx) => {
      await tx.$executeRaw`SELECT set_config('app.current_user_id', ${userId || null}, TRUE)`
      await tx.$executeRaw`SELECT set_config('app.current_transaction_id', 'pg_current_xact_id()', TRUE)`

      return fn(tx)
    })
```